### PR TITLE
Fix disruption controller permissions to allow patching pod's status

### LINKED ISSUE #113578: Disruption controller fails to clean up a stale disruption condition [when alpha PodDisruptionConditions enabled]
### What happened?

The `status` of a stale pod disruption condition remains `True` after two minutes as the disruption controller fails to change it to `False` to disable the condition.

### What did you expect to happen?

The `status` of a stale pod disruption condition should be changed to `False` after two minutes.

### How can we reproduce it (as minimally and precisely as possible)?

0. cluster with Alpha `PodDisurptionConditions` feature gate enabled
1. `DisruptionTarget` is added without associated delete operation (it may happen if preemption delete call fails, but it can be simulated by API)
2. Wait 2min as after this time the condition is recognized as stale by disruption controller, as the pod is not terminating

### Anything else we need to know?

The patch of disruption controller fails with the following message in the logs:
`disruption.go:651] error syncing Pod default/job-mem-pressure-k85rr to clear DisruptionTarget condition, requeueing: failed to patch status "{\"metadata\":{\"uid\":\"0369be75-680c-4676-ba8e-ba5075cbf22e\"},\"status\":{\"$setElementOrder/conditions\":[{\"type\":\"DisruptionTarget\"},{\"type\":\"Initialized\"},{\"type\":\"Ready\"},{\"type\":\"ContainersReady\"},{\"type\":\"PodScheduled\"}],\"conditions\":[{\"lastTransitionTime\":\"2022-11-02T18:24:36Z\",\"message\":null,\"reason\":null,\"status\":\"False\",\"type\":\"DisruptionTarget\"}]}}" for pod "default"/"job-mem-pressure-k85rr": pods "job-mem-pressure-k85rr" is forbidden: User "system:serviceaccount:kube-system:disruption-controller" cannot patch resource "pods/status" in API group "" in the namespace "default"`

### Kubernetes version

<details>
1.25.x
<!--```console
$ kubectl version
```
-->

</details>


### Cloud provider

<details>
Reproducible on KIND.
</details>


### OS version

<details>

</details>


### Install tools

<details>

</details>


### Container runtime (CRI) and version (if applicable)

<details>

</details>


### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:

Fixes #113578

#### Special notes for your reviewer:

The issue and the fix are very similar to this one: https://github.com/kubernetes/kubernetes/issues/112517.
There are no tests added in this PR. The permissions for disruption controller are tested by the `policy_test.go` unit test, but only for the feature gates enabled by default. The test asserts in the `controller-role-bindings.yaml` will be adjusted when the feature gradutes to Beta in the PR: https://github.com/kubernetes/kubernetes/pull/113360

#### Does this PR introduce a user-facing change?

```release-note
Fix that disruption controller changes the status of a stale disruption condition after 2 min when the PodDisruptionConditions feature gate is enabled
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy.go [hunk 1]
```diff
@@ -124,19 +124,25 @@ func buildControllerRoles() ([]rbacv1.ClusterRole, []rbacv1.ClusterRoleBinding)
...
+				rbacv1helpers.NewRule("get", "list", "watch").Groups(legacyGroup).Resources("replicationcontrollers").RuleOrDie(),
+				rbacv1helpers.NewRule("get", "list", "watch").Groups(policyGroup).Resources("poddisruptionbudgets").RuleOrDie(),
+				rbacv1helpers.NewRule("get", "list", "watch").Groups(appsGroup).Resources("statefulsets").RuleOrDie(),
+				rbacv1helpers.NewRule("update").Groups(policyGroup).Resources("poddisruptionbudgets/status").RuleOrDie(),
+				rbacv1helpers.NewRule("get").Groups("*").Resources("*/scale").RuleOrDie(),
+				eventsRule(),
+			},
+		}
+		if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+			role.Rules = append(role.Rules, rbacv1helpers.NewRule("patch").Groups(legacyGroup).Resources("pods/status").RuleOrDie())
```
Comment (liggitt) [inline plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy.go:142]: I'm a little surprised to see this added to a controller that did not have pod delete permissions... I thought the conditions were being added by the actors that were going to delete the pod

Comment (mimowo) [reply plugin/pkg/auth/authorizer/rbac/bootstrappolicy/controller_policy.go:142]: This controller is meant to disable a stale pod condition, It changes the status to `False` if the pod is non terminating (there was no successful delete) within 2min. This is a best effort tool that aims to resolve the issue that some in some situations, when the delete request fails, for example during preemption, then the condition might be left misleading the job's pod failure policy. However, the disruption controller does not guarantee removal as the pod may terminate within 2min for whatever reason.

**Comment by alculquicondor** [comment]
/lgtm

This also affects 1.25, right?

**Comment by mimowo** [comment]
> /lgtm
> 
> This also affects 1.25, right?

Right, but only in Alpha so I don't think it requires a cherry-pick.

**Pull request merged**